### PR TITLE
Fix NullReferenceException

### PR DESCRIPTION
**Pull request by pengweiqhca**

### CODE REVIEW

File: src/Elastic.Transport/DefaultHttpTransport.cs [hunk 1]
```diff
@@ -185,28 +185,28 @@ public DefaultHttpTransport(TConfiguration configurationValues) : this(configura
 			Settings.OnRequestDataCreated?.Invoke(requestData);
 			TResponse response = null;
 
-			if (OpenTelemetry.ElasticTransportActivitySource.HasListeners() && activity.IsAllDataRequested)
+			if (activity is { IsAllDataRequested: true })
```
Comment (Mpdreamz) [inline src/Elastic.Transport/DefaultHttpTransport.cs:188]:
```suggestion
			if (OpenTelemetry.ElasticTransportActivitySource.HasListeners() && activity is { IsAllDataRequested: true })
```

**Review by Mpdreamz — CHANGES_REQUESTED** [review]
Thank you for highlighting this! I have added one code suggestion.

File: src/Elastic.Transport/DefaultHttpTransport.cs (hunk 1 above)
Comment (pengweiqhca) [reply src/Elastic.Transport/DefaultHttpTransport.cs:188]: This is not necessary. If `OpenTelemetry.ElasticTransportActivitySource.HasListeners()` is false, `activity` must be null.

Comment (Mpdreamz) [reply src/Elastic.Transport/DefaultHttpTransport.cs:188]: Ahh missed the earlier assignment to `activity`

**Review by Mpdreamz — APPROVED** [review]

**Pull request merged**